qiskit/algorithms/minimum_eigensolvers/numpy_minimum_eigensolver.py

```diff
@@ -0,0 +1,113 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The NumPy minimum eigensolver algorithm and result."""
+
+from __future__ import annotations
+
+from typing import Callable
+import logging
+import numpy as np
+
+from qiskit.opflow import PauliSumOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.utils.deprecation import deprecate_function
+
+# TODO this path will need updating
+from ..eigen_solvers.numpy_eigen_solver import NumPyEigensolver
+from .minimum_eigensolver import MinimumEigensolver, MinimumEigensolverResult
+from ..list_or_dict import ListOrDict
+
+logger = logging.getLogger(__name__)
+
+
+class NumPyMinimumEigensolver(MinimumEigensolver):
+    """
+    The NumPy minimum eigensolver algorithm.
+    """
+
+    @deprecate_function(
+        "The NumPyMinimumEigensolver class has been superseded by the "
+        "qiskit.algorithms.minimum_eigensolvers.NumPyMinimumEigensolver class. "
+        "This class will be deprecated in a future release and subsequently "
+        "removed after that.",
+        category=PendingDeprecationWarning,
+    )
```

Comment:
Are we going to leave this drprecation message here somewhat as a signal that this class needs to be updated after VQD one is merged. Seems kinda odd to me. It should just be the import statement needs switching to use the new eigensolver one right? For now even using the old one is fine too since its not going away just yet!